docs-src/typescript/how-to-handle-workflow-logic-requirements-in-typescript.md

```diff
@@ -0,0 +1,91 @@
+---
+id: how-to-handle-workflow-logic-requirements-in-typescript
+title: How to handle Workflow logic requirements in TypeScript
+sidebar_label: Workflow logic requirements
+description: Handle Signal
+tags:
+  - developer-guide
+  - sdk
+  - typescript
+---
+
+In the Temporal TypeScript SDK, Workflows run in a deterministic sandboxed environment. The code is bundled on Worker creation using Webpack, and any package can be imported as long as it does not reference Node or DOM APIs.
+
+Since the Workflow sandbox can only run deterministic code, [Side Effects](/workflows#side-effect) and access to external state must be done through Activities. This limitation also means that Workflow code cannot directly import the Activity Definition. Activity Types may be imported, so they can be invoked in a type safe manner.
```

Comment:
```suggestion
Because the Workflow sandbox can run only deterministic code, [Side Effects](/workflows#side-effect) and access to external state must be done through Activities.
This limitation also means that Workflow code cannot directly import the Activity Definition.
Activity Types can be imported, so they can be invoked in a type-safe manner.
```